gitea: new portgroup

herbygillot opened the pull request:
#### Description

This PR seeks to add a new portgroup for projects hosted in [Gitea](https://gitea.com).
Being that you can run a private Gitea instance yourself, the portgroup potentially supports using custom domains via `gitea.domain`, but this hasn't been tested.

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
printf "%s\n" "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)" "$(xcodebuild -version|awk 'NR==1{x=$0}END{print x" "$NF}')"|tee /dev/tty|pbcopy
-->
macOS 10.15.7 19H114
Xcode 12.4 12D4e

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -d install`?
- [ ] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

herbygillot committed:
gitea: new portgroup, new port gitea-tea

- add support for gitea.com domain to golang portgroup

herbygillot commented:
CI likely failing because no ports have been changed here.

harens commented:
@herbygillot this PR has been open for quite a while now, and just looking through it, it looks good to me. 👍

Maybe we could add the PG to one of the gitea ports in this PR, just to show how it works? Then I think we can definitely merge this. Thanks for adding the PG @herbygillot.

pmetzger commented:
I think we should simply merge this. As nothing yet depends on it, we can always fix it later to be better if it turns out that it is somehow inadequate. My only complaint is that there should also be documentation added to the guide about this, but that can be added in a subsequent merge request.

herbygillot commented:
Will update this one shortly.

herbygillot commented:
Update here:

- added gitea.com support to the **golang** portgroup provided by this **gitea** portgroup
- added a port, `gitea-tea`, using the `golang` portgroup against the `gitea.com` domain

pmetzger commented:
Again, no need to delay merging this, but could you add comments to the portgroup and macports guide updates, perhaps in a separate merge request?

herbygillot commented:
OK, additional comments have been added to the `gitea` portgroup.  Will need to add `macports-guide` documentation separately.

pmetzger commented:
LGTM.

herbygillot commented:
Thank you @harens, @pmetzger.
